src/modclub.mo

```diff
@@ -88,35 +124,103 @@ module {
     // Proof of Humanity APIs
     verifyHumanity: (Text) -> async PohVerificationResponsePlus;
     subscribePohCallback: (SubscribePohMessage) -> async ();
   };
 
-  public let MODCLUB_CANISTER_ID_DEV = "hvyqe-cyaaa-aaaah-qdbiq-cai";
-  public let MODCLUB_DEV_ACTOR = actor "hvyqe-cyaaa-aaaah-qdbiq-cai" : ModclubActorType;
+  public type ModclubLedgerActorType = actor {
+    icrc1_balance_of : (Account) -> async Tokens;
+    icrc1_fee : () -> async Nat;
+    icrc1_decimals : () -> async Nat8;
+    icrc1_transfer : ({
+      from_subaccount : ?Subaccount;
+      to : Account;
+      amount : Tokens;
+      fee : ?Tokens;
+      memo : ?Memo;
+      created_at_time : ?Timestamp;
+    }) -> async Result<TxIndex, TransferError>;
+  };
 
+  public let MODCLUB_CANISTER_ID_QA = "hvyqe-cyaaa-aaaah-qdbiq-cai";
+  public let MODCLUB_CANISTER_ID_DEV = "d7isk-4aaaa-aaaah-qdbsa-cai";
   public let MODCLUB_CANISTER_ID_PROD = "gwuzc-waaaa-aaaah-qdboa-cai";
-  public let  MODCLUB_PROD_ACTOR =  actor "gwuzc-waaaa-aaaah-qdboa-cai" : ModclubActorType;
 
-  public func getModclubId(environment: Text) : Text {
-    if(environment == "prod") {
-      return MODCLUB_CANISTER_ID_PROD;
-    } else {
-      return MODCLUB_CANISTER_ID_DEV; 
+  public let MODCLUB_LEDGER_QA = "vckh6-hqaaa-aaaah-qc7wa-cai";
+  public let MODCLUB_LEDGER_DEV = "vxnwt-gyaaa-aaaah-qc7vq-cai";
+  public let MODCLUB_LEDGER_PROD = "xsi2v-cyaaa-aaaaq-aabfq-cai";
+
+  public func getModclubId(env: Text) : Text {
+    switch(env) {
+      case("prod") { MODCLUB_CANISTER_ID_PROD; };
+      case("dev") { MODCLUB_CANISTER_ID_DEV; };
+      case("qa") { MODCLUB_CANISTER_ID_QA; };
+      case(_) {"2vxsx-fae"}; // anonymous
     };
   };
 
-  public func getModclubActor(environment: Text) : ModclubActorType {
-    if(environment == "prod") {
-      return MODCLUB_PROD_ACTOR;
-    } else {
-      return MODCLUB_DEV_ACTOR;
+  public func getModclubLedgerId(env: Text) : Text {
+    switch(env) {
+      case("prod") { MODCLUB_LEDGER_PROD; };
+      case("dev") { MODCLUB_LEDGER_DEV; };
+      case("qa") { MODCLUB_LEDGER_QA; };
+      case(_) {"2vxsx-fae"}; // anonymous
     };
   };
+
+  public func getModclubActor(env: Text) : ModclubActorType {
+     actor (getModclubId(env)) : ModclubActorType;
+  };
+
+  public func getModclubLedgerActor(env: Text) : ModclubLedgerActorType {
+     actor (getModclubLedgerId(env)) : ModclubLedgerActorType;
+  };
+
+  public func initProvider(env : Text, name : Text, description : Text, ?companyLogo : ?Image) : async Result.Result<Bool, Text> {
+    let modclub = getModclubActor(env);
+    let resp = await modclub.registerProvider(name, description, ?companyLogo);
+    if (Text.contains("Registration successful", #text resp) or Text.contains("already registered", #text resp)) {
+      return #ok(true);
+    };
+    if (Text.contains("not in allow list", #text resp)) {
+      throw Error.reject("Unable to register provider. Provider is not in whitelist.");
```

Comment:
I think you can remove this `throw Error` line, otherwise the line below will never execute.